x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -430,27 +425,47 @@ private SSLContextHolder createSslContext(X509ExtendedKeyManager keyManager, X50
      * Parses the settings to load all SSLConfiguration objects that will be used.
      */
     Map<SSLConfiguration, SSLContextHolder> loadSSLConfigurations() {
-        Map<SSLConfiguration, SSLContextHolder> sslConfigurations = new HashMap<>();
-        sslConfigurations.put(globalSSLConfiguration, createSslContext(globalSSLConfiguration));
+        Map<SSLConfiguration, SSLContextHolder> sslContextHolders = new HashMap<>();
+        sslContextHolders.put(globalSSLConfiguration, createSslContext(globalSSLConfiguration));
+        this.sslConfigurations.put("_global", globalSSLConfiguration);
 
         final Settings transportSSLSettings = settings.getByPrefix(XPackSettings.TRANSPORT_SSL_PREFIX);
-        List<Settings> sslSettingsList = new ArrayList<>();
-        sslSettingsList.add(getHttpTransportSSLSettings(settings));
-        sslSettingsList.add(settings.getByPrefix("xpack.http.ssl."));
-        sslSettingsList.addAll(getRealmsSSLSettings(settings));
-        sslSettingsList.addAll(getMonitoringExporterSettings(settings));
-
-        sslSettingsList.forEach((sslSettings) ->
-                sslConfigurations.computeIfAbsent(new SSLConfiguration(sslSettings, globalSSLConfiguration), this::createSslContext));
+        Map<String, Settings> sslSettingsMap = new HashMap<>();
+        sslSettingsMap.put(XPackSettings.HTTP_SSL_PREFIX, getHttpTransportSSLSettings(settings));
+        sslSettingsMap.put("xpack.http.ssl", settings.getByPrefix("xpack.http.ssl."));
+        sslSettingsMap.putAll(getRealmsSSLSettings(settings));
+        sslSettingsMap.putAll(getMonitoringExporterSettings(settings));
+
+        sslSettingsMap.forEach((key, sslSettings) -> {
+            if (sslSettings.isEmpty()) {
+                storeSslConfiguration(key, globalSSLConfiguration);
+            } else {
+                final SSLConfiguration configuration = new SSLConfiguration(sslSettings, globalSSLConfiguration);
+                storeSslConfiguration(key, configuration);
+                sslContextHolders.computeIfAbsent(configuration, this::createSslContext);
+            }
+        });
 
         // transport is special because we want to use a auto-generated key when there isn't one
```

Comment:
can you do me a favor and delete this comment? It is a left over from me that is no longer relevant